OpenRA.Mods.Common/Traits/World/RenderPostProcessPassBase.cs

```diff
@@ -0,0 +1,55 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public abstract class RenderPostProcessPassBase : IRenderPostProcessPass
+	{
+		readonly Renderer renderer;
+		readonly IShader shader;
+		readonly IVertexBuffer<RenderPostProcessPassVertex> buffer;
+		readonly PostProcessPassType type;
+
+		protected RenderPostProcessPassBase(string name, PostProcessPassType type)
+		{
+			this.type = type;
+			renderer = Game.Renderer;
+			shader = renderer.CreateShader(new RenderPostProcessPassShaderBindings(name));
+			var vertices = new RenderPostProcessPassVertex[]
+			{
+				new(-1, -1),
+				new(1, -1),
+				new(1, 1),
+				new(1, 1),
+				new(-1, 1),
+				new(-1, -1)
+			};
+
+			buffer = renderer.CreateVertexBuffer<RenderPostProcessPassVertex>(6);
+			buffer.SetData(ref vertices, 6);
+		}
+
+		PostProcessPassType IRenderPostProcessPass.Type => type;
+		bool IRenderPostProcessPass.Enabled => Enabled;
```

Comment:
```suggestion
		public PostProcessPassType Type { get; }
		public abstract bool Enabled { get; }
```

There's no need to use multiple variables here

Reply:
They are needed because of `[RequireExplicitImplementation]`, which I have now remembered to add.

Reply:
It's why I dislike that property